fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabsStrip.kt

```diff
@@ -0,0 +1,371 @@
+                }
+            }
+        }
+    }
+}
+
+private fun LazyListState.isItemPartiallyVisible(itemInfo: LazyListItemInfo?) =
+    itemInfo != null &&
+        (itemInfo.offset + itemInfo.size > layoutInfo.viewportEndOffset || itemInfo.offset < 0)
+
+@Composable
+private fun TabItem(
+    state: TabStripItem,
+    modifier: Modifier = Modifier,
+    onCloseTabClick: (id: String) -> Unit,
+    onSelectedTabClick: (id: String) -> Unit,
+) {
+    TabStripCard(
+        modifier = modifier.fillMaxSize(),
+        backgroundColor =
+        if (state.isPrivate) {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer3
+            } else {
+                FirefoxTheme.colors.layer2
+            }
+        } else {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer2
+            } else {
+                FirefoxTheme.colors.layer3
+            }
+        },
+        elevation = if (state.isSelected) {
+            selectedTabStripCardElevation
+        } else {
+            defaultTabStripCardElevation
+        },
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxSize()
+                .clickable { onSelectedTabClick(state.id) },
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween,
+        ) {
+            Row(
+                modifier = Modifier.weight(1f, fill = false),
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                Spacer(modifier = Modifier.size(8.dp))
+
+                TabStripIcon(state.icon)
+
+                Spacer(modifier = Modifier.size(8.dp))
+
+                Text(
+                    text = state.title,
+                    color = FirefoxTheme.colors.textPrimary,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    style = FirefoxTheme.typography.subtitle2,
+                )
+            }
+
+            IconButton(onClick = { onCloseTabClick(state.id) }) {
+                Icon(
+                    painter = painterResource(R.drawable.mozac_ic_cross_20),
+                    modifier = Modifier.size(20.dp),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                    contentDescription = stringResource(R.string.close_tab),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun TabStripIcon(icon: Bitmap?) {
+    if (icon != null) {
+        Image(
+            bitmap = icon.asImageBitmap(),
+            contentDescription = null,
+            modifier = Modifier
+                .size(tabStripIconSize)
+                .clip(CircleShape),
+        )
+    } else {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_globe_24),
+            modifier = Modifier.size(tabStripIconSize),
+            tint = FirefoxTheme.colors.iconPrimary,
+            contentDescription = null,
+        )
```

Comment:
if we're wanting to be consistent, [we're using a FavIcon as the fallback in the Tabs Tray](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/TabThumbnail.kt#74-90)

Reply:
Nice! Thanks! I wanted to look into this, also now we don't need the bitmap icon